lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_viewmodel.dart

```diff
@@ -356,6 +359,41 @@ class WorkGroupNodesSurfingViewModel extends BaseViewModel {
     };
   }
 
+  OnlineThunkAction downloadNodeAction(List<WorkGroupNode> nodes, {ItemSelectionType itemSelectionType = ItemSelectionType.single}) {
+    return OnlineThunkAction((Store<AppState> store) async {
+      final status = await Permission.storage.status;
+      switch (status) {
+        case PermissionStatus.granted: _downloadNodes(nodes, itemSelectionType: itemSelectionType);
+        break;
+        case PermissionStatus.permanentlyDenied:
+          _appNavigation.popBack();
+          break;
+        default: {
+          final requested = await Permission.storage.request();
+          switch (requested) {
+            case PermissionStatus.granted: _downloadNodes(nodes, itemSelectionType: itemSelectionType);
+            break;
+            default: _appNavigation.popBack();
+            break;
+          }
+        }
+      }
+    });
+  }
+
+  void _downloadNodes(List<WorkGroupNode> nodes, {ItemSelectionType itemSelectionType = ItemSelectionType.single}) {
+    store.dispatch(_handleDownloadNodes(nodes.map((node) => node.workGroupNodeId)));
```

Comment:
I think you are missing this. We have to cancel multi selection on UI after they have pressed download button
`if (itemSelectionType == ItemSelectionType.multiple) {
      cancelSelection();
    }
`